### PR TITLE
[#165] manual regional maxMag

### LINKED ISSUE #165: Feature: Make regional Maximum Magnitude an API argument
SRM team want to experiment with forcing a upper bound on the maxMagnitude for Regional MFD. 

**NB** The current model (U3) is to calculate the maximum magnitude from the actual fault system ruptures (+ scaling relation)

Now we should take the user-defined max mag, and force the target for bins above max_mag to 1e-20.

### PR DESCRIPTION
**Pull request by voj**
closes #165

### CODE REVIEW

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_SubductionInversionTargetMFDs.java [hunk 1]
```diff
@@ -136,7 +136,7 @@ public NZSHM22_SubductionInversionTargetMFDs(NZSHM22_InversionFaultSystemRuptSet
 //		List<MFD_InversionConstraint> mfdUncertaintyConstraints = new ArrayList<>();
 
 		if (mfdUncertaintyWeightedConstraintWt > 0.0) {
-			mfdUncertaintyConstraints.add(MFDManipulation.addMfdUncertainty(targetOnFaultSupraSeisMFD, MINIMIZE_RATE_BELOW_MAG, mfdUncertaintyWeightedConstraintPower));
+			mfdUncertaintyConstraints.add(MFDManipulation.addMfdUncertainty(targetOnFaultSupraSeisMFD, MINIMIZE_RATE_BELOW_MAG, 20, mfdUncertaintyWeightedConstraintPower));
```
Comment (chrisdicaprio) [inline src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_SubductionInversionTargetMFDs.java:139]: @voj wondering about hard coded 20

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/CrustalMFDRunner.java [hunk 2]
```diff
@@ -109,7 +109,7 @@ public CrustalMFDRunner setOutputPath(String path) {
     }
 
     public void run() throws IOException {
-        NZSHM22_CrustalInversionTargetMFDs targetMfds = new NZSHM22_CrustalInversionTargetMFDs(rupSet, totalRateM5_Sans, totalRateM5_TVZ, bValue_Sans, bValue_TVZ, minMagSans, minMagTvz, mfdUncertaintyPower);
+        NZSHM22_CrustalInversionTargetMFDs targetMfds = new NZSHM22_CrustalInversionTargetMFDs(rupSet, totalRateM5_Sans, totalRateM5_TVZ, bValue_Sans, bValue_TVZ, minMagSans, minMagTvz, 15, 15, mfdUncertaintyPower);
```
[the remark this answers is not in the document]
Comment (voj) [reply src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/CrustalMFDRunner.java:112]: It's hardcoded to be outside the range, i.e. no change to before. This run() function is obsolete with changes I've got on another branch, so I just did a quick fix

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_SubductionInversionTargetMFDs.java (hunk 1 above)
Comment (voj) [reply src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_SubductionInversionTargetMFDs.java:139]: hardcoded to affect no change to before. My plan is to see if this change is what works for the scientists before fixing subduction

**Review by chrisbc — APPROVED** [review]
Thanks @voj

File: src/integration/java/nz/cri/gns/NZSHM22/opensha/hazard/NZSHM22_HazardCalculatorBuilder_IntegrationTest.java [hunk 3]
```diff
@@ -37,7 +37,7 @@ public void hazardCalcTest() throws URISyntaxException, DocumentException, IOExc
...
+        expected = Arrays.asList(1.0, 1.0, 1.0, 1.0, 1.0, 0.9999999999999999, 0.9999999999999998, 0.9999999999999989, 0.9999999999999925, 0.9999999999999312, 0.9999999999991339, 0.9999999999879503, 0.999999999831085, 0.9999999973983922, 0.9999999665851131, 0.9999996118919805, 0.9999962081168284, 0.9999699607181228, 0.9998081657023239, 0.9990289210478789, 0.9960411527590377, 0.9869816637761328, 0.9648683398418673, 0.9207806104979249, 0.8474196480157911, 0.7442864556904661, 0.6189901266372444, 0.48565138535305696, 0.3592749941799578, 0.25111598411936187, 0.16626604849636029, 0.10467524520049287, 0.06284799790336593, 0.03608342967890055, 0.019846907568932592, 0.010466902673227918, 0.005293722394825617, 0.0025657456413076263, 0.0011898271220269763, 5.266219848827358E-4, 2.2172907727058266E-4, 8.845264707091349E-5, 3.328593366547228E-5, 1.1760905246904763E-5, 3.884275600607445E-6, 1.194272810312036E-6, 3.4065684717532463E-7, 8.991112887546393E-8, 2.192300252445989E-8, 4.936000164157406E-9, 1.0269930461603849E-9);
```
Comment (chrisbc) [inline src/integration/java/nz/cri/gns/NZSHM22/opensha/hazard/NZSHM22_HazardCalculatorBuilder_IntegrationTest.java:40]: As per @voj chat File loading in old form was wrong, some lingering U3 modular artifacts.

**Review by chrisbc — APPROVED** [review]

**Pull request closed without merging**